.github/workflows/build-revive-wasm.yml

```diff
@@ -26,30 +25,43 @@ jobs:
         uses: actions/cache@v3
         with:
           path: |
-            target-llvm
+            target-llvm/emscripten/target-final
+            target-llvm/gnu/target-final
           # Use a unique key based on LLVM version or configuration files to avoid cache invalidation
-          key: llvm-build-${{ runner.os }}-${{ hashFiles('LLVM.lock', '.github/trigger-wasm-llvm-build') }}
+          key: llvm-build-${{ runner.os }}-${{ hashFiles('LLVM.lock', 'Cargo.toml', 'Cargo.lock', 'crates/llvm-builder/**', '.github/workflows/build-revive-wasm.yml') }}
 
-      - name: Install Dependencies
+      - name: Install system dependencies
         run: |
-          # system dependencies
           sudo apt-get update && sudo apt-get install -y cmake ninja-build curl git libssl-dev pkg-config clang lld
-          rustup target add wasm32-unknown-emscripten
-          # host LLVM
-          curl -sSL --output llvm.tar.xz https://github.com/paritytech/revive/releases/download/v0.1.0-dev.7/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-24.04.tar.xz
-          mkdir llvm18
-          tar Jxf llvm.tar.xz -C llvm18/
-          echo "LLVM_SYS_181_PREFIX=$(pwd)/llvm18" >> $GITHUB_ENV
-          # Clone LLVM and install the Emscripten SDK
+
+      - name: Install Rust stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          components: rust-src
+          target: wasm32-unknown-emscripten
+
+      - name: Install LLVM build dependencies
+        run: |
           make install-llvm-builder
           revive-llvm --target-env emscripten clone
```

Comment:
Shouldn't this also be guarded with `steps.cache-llvm.outputs.cache-hit != 'true'`?

Reply:
No; `revive-llvm --target-env emscripten clone` clones LLVM and the Emscripten SDK. The Emscripten SDK is required when compiling revive to Wasm.

Reply:
That being said, `revive-llvm-builder` could be more fine-granular in that regard. I.e. rework the `clone` command into a `dependency` command, which has more granular control over what it install. Then, during the LLVM build, we would install both, but during the revive build only emsdk.

But that's entirely optional and if desired should be done in another PR.

Reply:
So we build compile revive-llvm just to clone Emscripten SDK?

Reply:
Yes. Note that it doesn't only clone but also install the right version.

Reply:
The `revive-llvm` utility itself could also be cached 🙂 But that should also be a follow-up; I want to merge this here ASAP to unblock pending PRs.